### DISCUSSION TITLE
Is possible to use only Edge functions to interact with supabase?

### DISCUSSION THREAD

**Discussion by fabiojansenbr** [supabase/supabase#25972] [Questions]
Personally, I don't like the idea of putting business rules and access to services directly in the frontend. So the idea is simple, using edge functions for actions such as registering users, logging in, obtaining and writing data to tables.
With this, I will have more control over the business rules and will not depend on inserting anything into my mobile frontend.

Is there any limitation? higher costs?

**Comment by ssudaraka** [comment]
Hi @fabiojansenbr, You can find the costs involved with Edge Functions on the Supabase pricing page - https://supabase.com/pricing. In the free plan, 500,000 invocations are included.

For registering users and logging in users, you can use a Supabase client library (e.g., JavaScript SDK—https://supabase.com/docs/reference/javascript/introduction) to interact with the Auth API without having to create your own Edge Functions. However, if you wish to use Edge Functions, that option is available, too. It's totally up to your discretion.
